implementations/rust/ockam/ockam_api/src/authenticator/credential_issuer/credential_issuer.rs

```diff
@@ -0,0 +1,87 @@
+use core::time::Duration;
+
+use crate::authenticator::AuthorityMembersRepository;
+use ockam::identity::models::{CredentialAndPurposeKey, CredentialSchemaIdentifier};
+use ockam::identity::utils::AttributesBuilder;
+use ockam::identity::{Attributes, Credentials, Identifier};
+use ockam_core::compat::sync::Arc;
+use ockam_core::Result;
+
+/// Legacy value, should be removed when all clients are updated to the latest version
+pub const TRUST_CONTEXT_ID: &[u8] = b"trust_context_id";
+
+/// Identifier for the schema of a project credential
+pub const PROJECT_MEMBER_SCHEMA: CredentialSchemaIdentifier = CredentialSchemaIdentifier(1);
+
+/// Maximum duration for a valid credential in seconds (30 days)
+pub const DEFAULT_CREDENTIAL_VALIDITY: Duration = Duration::from_secs(30 * 24 * 3600);
+
+/// This struct runs as a Worker to issue credentials based on a request/response protocol
+pub struct CredentialIssuer {
```

Comment:
It would be good to add a comment explaining how each part of the `CredentialIssuer` is used (or comment directly on the attributes definitions)